src/app/transformers/angular-material.transformer.ts

```diff
@@ -31,16 +35,33 @@ export class AngularMaterialTransformer implements Transformer {
 
     renderTextControl(key) {
         let result;
+        const validations = this.settings.find(e => e.key === 'validations');
 
-        result = `
+        if (validations.value) {
+
+            result = `
 
     <div>
         <mat-form-field>
             <input matInput placeholder="${key}" id="${key}"
-            ngModel name="${key}" #${key}="ngModel" required>
+            [(ngModel)]="${key}" name="${key}"  [formControl]="${key}FormControl">
+            <mat-error *ngIf="${key}FormControl.invalid">

```

Comment:
Looks great, but according to [material documentation](https://material.angular.io/components/input/overview#error-messages), when there is only a single `<mat-error>` there is no need for the `*ngIf`.

But then again - I don't have a lot of experience with material design, so what ever you think is best...

Reply:
It depends on how the user implement the validation checkings, therefore the *ngIf may not be a Must but it can be useful for the user...

Reply:
Sounds good to me, Thank you!